src/components/SuggestionsList/SuggestionsList.tsx

```diff
@@ -0,0 +1,158 @@
+  loadingText: string;
+  /**
+   * Text to show when the results are empty for the active search.
+   */
+  noResultsText: string;
+  /**
+   * The groups of items to render.
+   */
+  groupedItems?: SuggestionItemGroupProps[];
+  /**
+   * The index of current keyboard selection.
+   */
+  selectedIndex: number;
+  /**
+   * The active search that produced the current state.
+   */
+  search: string;
+  /**
+   * Called when an item in a group is selected by clicking.
+   */
+  onItemSelected(item: SuggestionItem): void;
+}
+
+interface SuggestionsListWithResultsProps extends SuggestionsListProps {
+  groupedItems: SuggestionItemGroupProps[];
+}
+
+export interface SuggestionItemGroupProps {
+  title: string;
+  items: SuggestionItem[];
+}
+
+export interface SuggestionsListState {
+  hoveredId: string | number | null;
+}
+
+const hasResults = (props: SuggestionsListProps): props is SuggestionsListWithResultsProps => {
+  return !!props.groupedItems && props.groupedItems.length > 0;
+};
+
+/**
+ * A `SuggestionsList` displays a list of search results in a dropdown.
+ */
+export default class SuggestionsList extends React.PureComponent<
+  SuggestionsListProps,
+  SuggestionsListState
+> {
+  public state: SuggestionsListState = { hoveredId: null };
+
+  public render() {
+    const results = this.getResults();
+    const { left, top } = this.props;
+    return (
+      <div className="y-suggestions-list" style={{ left, top }}>
+        {results}
+      </div>);
+  }
+
+  private getResults() {
+    const statusSection = this.getSearchStatusSection();
+    if (!hasResults(this.props)) {
+      return statusSection;
+    }
+    return (
+      <span>
+        <ul className="y-suggestions-list--results">
+          {this.props.groupedItems.map(this.getGroupResults)}
+        </ul>
+        {statusSection}
+      </span>);
+  }
+
+  private getGroupResults = (group: SuggestionItemGroupProps) => {
+    const items = group.items.map(this.getResultItem);
+    return (
+      <li key={group.title}>
+        <ul onMouseOut={this.onMouseOut}>
+          {items}
+        </ul>
+      </li>);
+  }
+
+  private getResultItem = (item: SuggestionItem, index: number) => {
+    return (
+      <li key={item.id}>
+        <SuggestionsListItem
+          search={this.props.search}
+          isHovered={item.id === this.state.hoveredId}
+          isSelected={index === this.props.selectedIndex}
+          onHover={this.onHover}
+          onSelect={this.props.onItemSelected}
+          {...item}
+        />
+      </li>);
+  }
+
+  private getSearchStatusSection() {
+    const content = this.props.isLoading ? this.getLoading() : this.getNoResults();
+    return (
+      <span>
+        <Block bottomSpacing={GutterSize.XSMALL} />
+        <Block
+          padding={GutterSize.MEDIUM}
+          textAlign="center"
+          className="y-suggestions-list--status"
+        >
+          {content}
+        </Block>
+        <Block bottomSpacing={GutterSize.XSMALL} />
```

Comment:
@swese44 I can't get this to be the right size - according to @adrianmg's design this should be 48px tall

Reply:
ya the empty Blocks are ugly. i'd just add CSS where it's easier than wonky markup and we can see if we can build those options into Block etc. in the future